### PR TITLE
GEODE-3529 move new client/server security classes to a different pac…

### PR DESCRIPTION
**Pull request by bschuchardt**
@kohlmu-pivotal @hiteshk25 @galen-pivotal @pivotal-amurmann @metatype 

The new classes/interfaces have been moved to org.apache.geode.security.server.

A package.html is now provided that describes the purpose of the package.

The word "Stream" has been removed from the names of these classes.  Those
names make sense in the context of the Protobuf support classes in
org.apache.geode.internal.cache.tier.sockets classes but not when taken
as a separate group of classes.

The method Authenticator.receiveMessage() has been renamed to
Authenticator.authenticate().

Thank you for submitting a contribution to Apache Geode.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [ x] Is there a JIRA ticket associated with this PR? Is it referenced in the commit message?

- [x] Has your PR been rebased against the latest commit within the target branch (typically `develop`)?

- [x] Is your initial contribution a single, squashed commit?

- [x] Does `gradlew build` run cleanly?

- [x] Have you written or updated unit tests to verify your changes?

- [n/a] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?

### Note:
Please ensure that once the PR is submitted, you check travis-ci for build issues and
submit an update to your PR as soon as possible. If you need help, please send an
email to dev@geode.apache.org.

### CODE REVIEW

**Comment by bschuchardt** [comment]
Travis check failed in an unrelated unit test.  This test passes on my branch when I run it individually or as part of the "build" task.

**Comment by galen-pivotal** [comment]
@bschuchardt try closing and reopening; I think that will make Travis restart.

File: geode-core/src/main/java/org/apache/geode/internal/cache/tier/sockets/ServerConnectionFactory.java [hunk 1]
```diff
@@ -41,8 +41,8 @@ private synchronized void initializeAuthenticatorsMap() {
       return;
     }
     authenticators = new HashMap<>();
-    ServiceLoader<StreamAuthenticator> loader = ServiceLoader.load(StreamAuthenticator.class);
-    for (StreamAuthenticator streamAuthenticator : loader) {
+    ServiceLoader<Authenticator> loader = ServiceLoader.load(Authenticator.class);
+    for (Authenticator streamAuthenticator : loader) {
```
Comment (hiteshk25) [inline geode-core/src/main/java/org/apache/geode/internal/cache/tier/sockets/ServerConnectionFactory.java]: may we want to remove stream from variable name

File: geode-core/src/main/java/org/apache/geode/internal/cache/tier/sockets/ServerConnectionFactory.java [hunk 2]
```diff
@@ -57,15 +57,14 @@ private synchronized ClientProtocolMessageHandler initializeMessageHandler() {
     return protocolHandler;
   }
 
-  private StreamAuthenticator findStreamAuthenticator(String implementationID) {
+  private Authenticator findStreamAuthenticator(String implementationID) {
```
Comment (hiteshk25) [inline geode-core/src/main/java/org/apache/geode/internal/cache/tier/sockets/ServerConnectionFactory.java]: stream in method name

File: geode-core/src/main/java/org/apache/geode/internal/cache/tier/sockets/ServerConnectionFactory.java [hunk 3]
```diff
@@ -57,15 +57,14 @@ private synchronized ClientProtocolMessageHandler initializeMessageHandler() {
...
     if (authenticators == null) {
       initializeAuthenticatorsMap();
     }
-    Class<? extends StreamAuthenticator> streamAuthenticatorClass =
-        authenticators.get(implementationID);
+    Class<? extends Authenticator> streamAuthenticatorClass = authenticators.get(implementationID);
```
Comment (hiteshk25) [inline geode-core/src/main/java/org/apache/geode/internal/cache/tier/sockets/ServerConnectionFactory.java]: stream in variable name

File: geode-core/src/main/java/org/apache/geode/internal/cache/tier/sockets/ServerConnectionFactory.java (hunk 1 above)
Comment (bschuchardt) [reply geode-core/src/main/java/org/apache/geode/internal/cache/tier/sockets/ServerConnectionFactory.java]: The "stream" concept is fine in the cache.tier.sockets package - there are other uses of it there.

**Comment by bschuchardt** [comment]
okay - travis ci passed this time

**Pull request closed without merging**